File: JavaScript/romanToInteger.js


/**
 * 
 * @param {string} s 
 */
const romanToInteger = (s) => {
  let curr_val; 
  let result = 0; 

  let num_mapper = {
    'I': 1, 
    'V': 5, 
    'X': 10, 
    'L': 50,
    'C': 100, 
    'D': 500, 
    'M': 1000
  }; 

  for(let i =0; i<s.length; i++){
    curr_val = num_mapper[s[i]]; 

    if(i + 1 < curr_val && curr_val < num_mapper[s[i + 1]]){
      result -= curr_val; 
    }else {
      result += curr_val; 
    }
  }

  return result; 
}; 

console.log(romanToInteger('XL'))

File: Java/RomanToInteger.java
import java.util.HashMap;
import java.util.Map;

class RomanToInt {
  public static int romanToInt(String s){
    int result = 0; 
    Map<Character, Integer> romanMap = new HashMap(); 

    romanMap.put('I', 1);
    romanMap.put('V', 5);
    romanMap.put('X', 10);
    romanMap.put('L', 50);
    romanMap.put('C', 100);
    romanMap.put('D', 500);
    romanMap.put('M', 1000); 

    for(int i = 0; i<s.length(); i++){
      int currVal = romanMap.get(s.charAt(i));

      if(i + 1 < s.length() && currVal < romanMap.get(s.charAt(i + 1))){
        result -= currVal;
      }else {
        result += currVal;
      }
    }


    return result;
  }

  public static void main(String[] args){
    int test = romanToInt("III");

    System.out.println(test);
  }
}